### PR TITLE
chore: remove deprecated update & upgrade commands

### LINKED ISSUE #7046: Remove deprecated update/upgrade commands
### Summary 

The CLI command deprecation list [here](https://jenkins-x.io/commands/deprecation/), lists the `update cluster`, `upgrade cluster` and `upgrade extensions` commands as deprecated. But we still support them within Jenkins X despite passing the deprecation date.

We should remove these deprecated commands.

### PR DESCRIPTION
**Pull request by mrageh**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
Remove deprecated update/upgrade commands.

#### Which issue this PR fixes
fixes #7046

### CODE REVIEW

File: pkg/cmd/update/update_cluster_gke.go [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-
-// CreateClusterOptions the flags for running create cluster
-type UpdateClusterGKEOptions struct {
-	UpdateClusterOptions
-}
-
-var (
-	updateClusterGKELong = templates.LongDesc(`
-		
-		Not currently implemented.
```
Comment (hferentschik) [inline pkg/cmd/update/update_cluster_gke.go:19]: LOL

File: pkg/cmd/upgrade/upgrade_extensions.go [hunk 2]
```diff
@@ -1,381 +0,0 @@
...
-
-	upgradeExtensionsExample = templates.Examples(`
-		
-		# upgrade extensions
-		jx upgrade extensions
-
-	`)
-)
-
-type UpgradeExtensionsOptions struct {
```
Comment (hferentschik) [inline pkg/cmd/upgrade/upgrade_extensions.go:57]: @abayer, @daveconde, @dgozalo, @ccojocar - is it safe to remove extension-related stuff? There are no "extensions" in use anymore, right?

File: pkg/cmd/update/update_cluster_gke.go (hunk 1 above)
Comment (mrageh) [reply pkg/cmd/update/update_cluster_gke.go:19]: 😆

**Pull request merged**